chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2193 @@
+    _aux_block = True
+    _tri_power = True
+    _gmrs = False
+    _ham = False
+    _mem_params = (0x1F2F)  # poweron_msg offset
+
+    # offset of fw version in image file
+    _fw_ver_file_start = 0x1838
+    _fw_ver_file_stop = 0x1846
+    NEEDS_COMPAT_SERIAL = False
+    _valid_chars = TDH8_CHARSET
+
+    @classmethod
+    def get_prompts(cls):
+        rp = chirp_common.RadioPrompts()
+        rp.experimental = \
+            ('Due to the fact that the manufacturer continues to '
+             'release new versions of the firmware with obscure and '
+             'hard-to-track changes, this driver may not work with '
+             'your device. Thus far and to the best knowledge of the '
+             'author, no TD-H8 radios have been harmed by using CHIRP. '
+             'However, proceed at your own risk!')
+        rp.pre_download = (dedent("""\
+            1. Turn radio off.
+            2. Connect cable to mic/spkr connector.
+            3. Make sure connector is firmly connected.
+            4. Turn radio on (volume may need to be set at 100%).
+            5. Ensure that the radio is tuned to channel with no activity.
+            6. Click OK to download image from device."""))
+        rp.pre_upload = (dedent("""\
+            1. Turn radio off.
+            2. Connect cable to mic/spkr connector.
+            3. Make sure connector is firmly connected.
+            4. Turn radio on (volume may need to be set at 100%).
+            5. Ensure that the radio is tuned to channel with no activity.
+            6. Click OK to upload image to device."""))
+        return rp
+
+    def get_features(self):
+        rf = chirp_common.RadioFeatures()
+        rf.has_settings = True
+        rf.has_bank = False
+        rf.has_cross = True
+        rf.has_ctone = True
+        rf.has_rx_dtcs = True
+        rf.has_tuning_step = False
+        rf.has_ctone = True
+        rf.can_odd_split = True
+        rf.valid_name_length = 8
+        rf.valid_characters = self._valid_chars
+        rf.valid_skips = []
+        rf.valid_tmodes = ["", "Tone", "TSQL", "DTCS", "Cross"]
+        rf.valid_cross_modes = [
+            "Tone->Tone",
+            "DTCS->",
+            "->DTCS",
+            "Tone->DTCS",
+            "DTCS->Tone",
+            "->Tone",
+            "DTCS->DTCS"]
+        rf.valid_power_levels = TX_POWER
+        rf.valid_duplexes = ["", "-", "+", "split", "Off"]
+        rf.valid_modes = ["FM", "NFM"]
+        rf.valid_tuning_steps = STEPS
+
+        normal_bands = [self._vhf_range, self._uhf_range]
+
+        if self._mmap is None:
+            rf.valid_bands = [normal_bands[0], normal_bands[1]]
+        else:
+            rf.valid_bands = normal_bands
+        rf.memory_bounds = (1, 199)
+        return rf
+
+    def process_mmap(self):
+        self._memobj = bitwise.parse(MEM_FORMAT, self._mmap)
+
+    def validate_memory(self, mem):
+        msgs = chirp_common.CloneModeRadio.validate_memory(self, mem)
```

Comment:
This still works, but the python3 way to do this is:
```
msgs = super().validate_memory(mem)
```

So when you move it, please make that change.